app/routes.py

```diff
@@ -1,2 +1,304 @@
+            response_body = {
+                'details': 'Invalid data'
+            }
+            return make_response(jsonify(response_body), 400)
+        # making an instance of task after entering info
+        new_task = Task(title=request_body["title"],
+            description=request_body["description"], 
+            completed_at=request_body["completed_at"])
+
+        # add andcommit newtask
+
+        db.session.add(new_task)
+        db.session.commit()
+        # return task to user
+        return {"task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete" : (False
+            if new_task.completed_at == None else True)
+            }},201
+    # get all tasks
+    elif request.method == "GET":
+        title_query = request.args.get("title")
+        if title_query:
+            tasks = Task.query.filter_by(title=title_query)
+        else:
+            tasks = Task.query.all()
+
+
+        sort_query = request.args.get("sort")
+        # account for the asc/desc in wave 2(?)
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()) 
+        else:
+            tasks = Task.query.all()
+
+        # dd each task to be returned
+        task_response = []
+        for task in tasks:
+            task_response.append({
+                        "id": task.task_id,
+                        "title": task.title,
+                        "description": task.description,
+                        "is_complete": (False
+            if task.completed_at == None else True)})
+        return jsonify(task_response)
+            
+        
+        
+@tasks_bp.route("/<task_id>",methods=["GET", "PUT","DELETE"])
+def handle_one_task(task_id):
+    task = Task.query.get(task_id)
+    # if taskdoesnt exist, give 404
+    if task is None:
+        return make_response(f"Task {task_id} not found", 404)
+
+    # acconting for the descrepancy in wave 6 where it was saying to implement goal_id in thisresponse body.
+    # fixed this by essentially "if there is a goal id at this task, then return that, if not, then return it without the goal id
+    elif request.method == "GET":
+        if not task.goal_id:
+            return {"task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete" : (False
+                if task.completed_at == None else True)
+                }},200
+        else:
+            return {"task": {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : (False
+            if task.completed_at == None else True)
+            }},200
```

Comment:
Notice how these conditionals return a dictionary with the same key value pairs (besides goal_id) ? Instead of explicitly writing out this same dictionary twice, we can write a "base" dictionary and add the goal_id to it if it exists for the task:


```suggestion
    task_dict = {
            "id": task.task_id,
            "title": task.title,
            "description": task.description,
            "is_complete": False if task.completed_at == None else True
        }

    if task.goal_id:
        task_dict["goal_id] = task.goal_id   
    return task_dict, 200
```